taskboot/cargo.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import logging
+import subprocess
+
+import taskcluster
+
+from taskboot.config import Configuration
+
+logger = logging.getLogger(__name__)
+
+
+def cargo_publish(target, args):
+    """
+    Publish a crate on crates.io
+    """
+
+    # Load config from file/secret
+    config = Configuration(args)
+    assert config.has_cargo_auth(), "Missing Cargo authentication"
+
+    # Publish a crate on crates.io
+    try:
+        subprocess.run(
+            ["cargo", "publish", "--token", config.cargo["token"]], check=True
+        )
```

Comment:
Is the token printed in the console?
If yes, we need to use `capture_output` to ensure it isn't.

Reply:
Yes, it shows the entire command with the token if an error occurs

Reply:
As dicussed on Matrix, it is not in Cargo's stderr, but it is in the exception raised by subprocess.
To fix it, we just need to remove `check=True`, manually check the return code ourselves and raise the custom exception if the return code is not 0.